thorlcr/activities/loop/thloopslave.cpp

```diff
@@ -304,12 +304,16 @@ class CLoopSlaveActivity : public CLoopSlaveActivityBase
             {
                 while (!abortSoon)
                 {
-                    OwnedConstThorRow ret = (void *)curInput->nextRow();
-                    if (!ret)
+                    OwnedConstThorRow ret;
                     {
-                        ret.setown(curInput->nextRow()); // more cope with groups somehow....
+                        LookAheadTimer t(slaveTimerStats, timeActivities);
```

Comment:
hm, I think it's a bit more nuanced than this.
curInput is only the inputStream initially.. it then changes, the 2nd iteration of the loop, is not processing the input, but the result of the previous iteration. As it stands if it's doing many iterations, it will count it all against look ahead time.
I think you need to set a flag and only do until curInput changes.